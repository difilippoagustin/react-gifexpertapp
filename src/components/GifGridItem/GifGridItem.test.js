import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "./GifGridItem";

describe("<GifGridItem />", () => {
  const titleProvided = "Title";
  const urlProvided = "https://exmple.com";
  const wrapper = shallow(
    <GifGridItem title={titleProvided} url={urlProvided} />
  );

  test("should display component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should display the title in p tag correctly", () => {
    const p = wrapper.find("p").text();
    expect(p).toBe(titleProvided);
  });
  test("should have the correct url", () => {
    const url = wrapper.find("img").prop("src");
    expect(url).toBe(urlProvided);
  });
  test("should have the correct alt", () => {
    const alt = wrapper.find("img").prop("alt");
    expect(alt).toBe(titleProvided);
  });
  test("should have the correct class", () => {
    const className = wrapper.find("div").hasClass("animate__fadeInDown");
    expect(className).toBeTruthy();
  });
});
