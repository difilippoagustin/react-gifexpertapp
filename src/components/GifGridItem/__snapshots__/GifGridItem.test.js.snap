// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> should display component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <p>
    Title
  </p>
  <img
    alt="Title"
    src="https://exmple.com"
  />
</div>
`;
